asv_bench/benchmarks/benchmarks_hindcast.py

```diff
@@ -0,0 +1,166 @@
+# Write the benchmarking functions here.
+# See "Writing benchmarks" in the asv docs for more information.
+
+
+import dask
+import numpy as np
+import xarray as xr
+
+from climpred.bootstrap import bootstrap_hindcast
+# from climpred.constants import HINDCAST_COMPARISONS
+from climpred.prediction import compute_hindcast
+
+from . import parameterized, randn, requires_dask
+
+HINDCAST_COMPARISONS = ['m2r']  # e2r and probabilistic dont match
+METRICS = ['rmse', 'pearson_r', 'crpss']
+
+bootstrap = 8
+
+
+def _ensure_loaded(res):
+    """Compute no lazy results."""
+    if dask.is_dask_collection(res):
+        res = res.compute()
+    return res
+
+
+class Generate:
+    """
+    Generate random hind, control to be benckmarked.
+    """
+
+    timeout = 600
+    repeat = (2, 5, 20)
+
+    def make_hind_ref(self):
+        """hind and ref mimik hindcast experiment"""
+        self.hind = xr.Dataset()
+        self.reference = xr.Dataset()
+        self.hist = xr.Dataset()
+
+        self.nmember = 3
+        self.nlead = 3
+        self.nx = 64
+        self.ny = 64
+        self.init_start = 1960
+        self.init_end = 2000
+        self.ninit = self.init_end - self.init_start
+
+        frac_nan = 0.0
+
+        inits = np.arange(self.init_start, self.init_end)
+        leads = np.arange(1, 1 + self.nlead)
+        members = np.arange(1, 1 + self.nmember)
+
+        lons = xr.DataArray(
+            np.linspace(0, 360, self.nx),
+            dims=('lon',),
+            attrs={'units': 'degrees east', 'long_name': 'longitude'},
+        )
+        lats = xr.DataArray(
+            np.linspace(-90, 90, self.ny),
+            dims=('lat',),
+            attrs={'units': 'degrees north', 'long_name': 'latitude'},
+        )
+        self.hind['tos'] = xr.DataArray(
```

Comment:
If using `foo units` and so on, maybe just make this generic. Like `foo` or `var` or something, since `tos` is unique to us oceanographers.